pkg/bootstrap/platform/gcp.go

```diff
@@ -230,26 +232,28 @@ func (e *gcpEnv) Labels() map[string]string {
 	var instanceLabels map[string]string
 	go func() {
 		// use explicit credentials with compute.instances.get IAM permissions
-		creds, err := google.FindDefaultCredentials(ctx, compute.ComputeReadonlyScope)
+		creds, err := google.FindDefaultCredentials(ctx, ComputeReadonlyScope)
 		if err != nil {
 			log.Warnf("failed to find default credentials: %v", err)
 			success <- false
 			return
 		}
-		computeService, err := compute.NewService(ctx, option.WithCredentials(creds))
+		url := fmt.Sprintf("https://compute.googleapis.com/compute/v1/projects/%s/zones/%s/instances/%s", md[GCPProject], md[GCPLocation], md[GCEInstance])
+		resp, err := oauth2.NewClient(ctx, creds.TokenSource).Get(url)
 		if err != nil {
-			log.Warnf("failed to create new service: %v", err)
+			log.Warnf("unable to retrieve instance labels: %v", err)
 			success <- false
 			return
+
 		}
-		// instance.Labels is nil if no labels are present
-		instanceObj, err := computeService.Instances.Get(md[GCPProject], md[GCPLocation], md[GCEInstance]).Do()
-		if err != nil {
-			log.Warnf("unable to retrieve instance: %v", err)
+		defer resp.Body.Close()
+		instance := &GcpInstance{}
+		if err := json.NewDecoder(resp.Body).Decode(instance); err != nil {
```

Comment:
Are you sure it's safe to use this instead of jsonpb? I think most of that repo comes from google3 proto sync.

Reply:
I am pretty sure this is what the library is doing internally. The endpoint just returns a normal json, I tested this out on my machine and it seems to work at least, although that is admittedly not a great bar of testing...